substrate/substrate-processor/src/processor/batchProcessor.ts

```diff
@@ -339,6 +339,46 @@ export class SubstrateBatchProcessor<Item extends {kind: string, name: string} =
         return this
     }
 
+    /**
+     * Similar to {@link .addEvent},
+     * but requests `Ethereum.Executed` events holding a result of EVM transaction
+     * with an option to filter them by contract address.
+     *
+     * @example
+     * // request all `Ethereum.Executed` events from contract `0x6a2d262D56735DbA19Dd70682B39F6bE9a931D98`
+     * processor.addEthereumExecuted('0x6a2d262D56735DbA19Dd70682B39F6bE9a931D98')
+     *
+     * // request the same data from multiple contracts at once
+     * processor.addEthereumExecuted([
+     *     '0x6a2d262D56735DbA19Dd70682B39F6bE9a931D98',
+     *     '0x3795C36e7D12A8c252A20C5a7B455f7c57b60283'
+     * ])
+     */
+    addEthereumExecuted(
```

Comment:
imo, It should be automatically added together with Ethereum.transact in addEthereumTransaction

Reply:
and maybe the archive should look for this event in Ethereum.transact extrinsic events

Reply:
so request will stay the same, but response will have additional event

Reply:
but maybe it will be not backward compatible change

Reply:
> imo, It should be automatically added together with Ethereum.transact in addEthereumTransaction

like an additional field for a data selection part?
```
addEthereumTransaction(address, {executed: {args: true}})
```

Reply:
hm, I've forgot that there is also dataselection part

Reply:
I also think that the rule should be:

>for each `Ethereum.transact` call in the result set, always include `Ethereum.executed` emitted by the call

No need to create additional data selection options. Nothing should be changed on the SDK side at all.

Reply:
But it's not obviius that there will be another event in archive response. In some cases in can crash squid

Reply:
there should be something like `ethereumExecuted: true`? to make it backward compatible

Reply:
> In some cases in can crash squid

In what cases? 

Mappings developer should never make assumptions about exact set of returned data. The only guarantee is that requested data (and fields) will be there.

Reply:
i am a bit afraid that at some moment `Ethereum.Executed` will be changed or just stop being responsible for transaction status...

extending `CallDataSelection` with `ethereumExecuted: true` causes extra code which i neither want to implement and support. so lets continue with implicit executed events related to requested transactions

Reply:
Anyway we need to improve addTransaction method, it should add both ethereum.transact and ethereum.executed items to processor type, otherwise code like
`
if (item.name === 'Ethereum.Executed')
`
will not work

Reply:
```
    addEthereumTransaction<R extends CallDataRequest>(
        contractAddress: string | string[],
        options: {range?: Range, sighash?: string} & DataSelection<R>
    ): SubstrateBatchProcessor<AddEventItem<AddCallItem<Item, CallItem<"Ethereum.transact", R>>>, EventItem<{event: {args: true}}>>
  ```
 maybe like this, but looks ugly